pkg/cache/nodes.go

```diff
@@ -100,10 +101,18 @@ func (nc *schedulerNodes) addAndReportNode(node *v1.Node, reportNode bool) {
 
 	// add node to nodes map
 	if _, ok := nc.nodesMap[node.Name]; !ok {
+		nodeLabels, err := json.Marshal(node.Labels)
```

Comment:
what if `node.Labels` is nil?
can we  also add a UT coverage for the case where node has no labels defined?

Reply:
hi @chenya-zhang can you pls also take a look at this comment pls? thx

Reply:
Thanks for the reminder @yangwwei! Sorry I failed to see this comment last time.

If node.Labels is nil, JSON marshal will encode it as the "null" JSON value. We will check string "null" when decoding it.
Our existing UT in https://github.com/apache/incubator-yunikorn-k8shim/blob/84f52b7d72cdac1bc4904cf5a631617070225d29/pkg/cache/nodes_test.go#L76 should cover the case if a node has no labels. 

On the other hand, your comment reminds me to double check what if we fail to marshal. Unfortunately, JSON marshal will return a nil along with an err. This nil will be problematic if passing to string() directly. In this case, I assign a 0 byte to the JSON format of node labels, which is an empty string for us to check in YK core.